### PR TITLE
TNO-1651: Search unpublished content

### PR DESCRIPTION
**Pull request by jdtoombs**
In this PR:

- advanced search now contains an option to search unpublished content
- small changes to content controller

search-unpub

### CODE REVIEW

File: app/editor/src/features/admin/reports/constants/defaultReport.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const defaultReport: IReportModel = {
       showSentiment: false,
     },
     content: {
+      clearFolders: false,
```
Comment (jdtoombs) [inline app/editor/src/features/admin/reports/constants/defaultReport.ts:26]: This was throwing an error for me, just want to confirm this is ok,

File: api/net/Areas/Subscriber/Controllers/ContentController.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public class ContentController : ControllerBase
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(IPaged<Services.Models.Content.ContentModel>), (int)HttpStatusCode.OK)]
     [SwaggerOperation(Tags = new[] { "Content" })]
-    public async Task<IActionResult> FindWithElasticsearchAsync()
+    public async Task<IActionResult> FindWithElasticsearchAsync([FromQuery] string? index)
```
Comment (Fosol) [inline api/net/Areas/Subscriber/Controllers/ContentController.cs:75]: Don't implement this way.  Instead add a boolean field to the ContentFilter.  This way we don't have to manage index names on the client app and have to deal with invalid index names.  I'll probably change the editor implementation at some point too.

File: app/editor/src/features/admin/reports/constants/defaultReport.ts (hunk 1 above)
Comment (Fosol) [reply app/editor/src/features/admin/reports/constants/defaultReport.ts:26]: yes thanks.  I was surprised it wasn't throwing an error for me.  I have it fixed in my current work.

File: app/subscriber/src/components/sidebar/advanced-search/interfaces/IAdvancedSearchFilter.ts [hunk 3]
```diff
@@ -21,4 +21,6 @@ export interface IAdvancedSearchFilter {
   hasFile?: boolean;
   /** whether  not to bold the keywords */
   boldKeywords?: boolean;
+  /** index to control whether content is published or unpublished */
+  index?: string;
```
Comment (Fosol) [inline app/subscriber/src/components/sidebar/advanced-search/interfaces/IAdvancedSearchFilter.ts:25]: Change to `useUnpublished: boolean`

**Review by Fosol — CHANGES_REQUESTED** [review]

File: api/net/Areas/Subscriber/Controllers/ContentController.cs [hunk 4]
```diff
@@ -77,11 +77,12 @@ public async Task<IActionResult> FindWithElasticsearchAsync()
         var uri = new Uri(this.Request.GetDisplayUrl());
         var query = Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery(uri.Query);
         var filter = new ContentFilter(query);
+        var useUnpublished = filter.UseUnpublished == true;
         if (filter.Actions.Any(x => x.Contains("Commentary")) && filter.Quantity == 10)
         {
             filter.Quantity = 500;
         }
-        var result = await _contentService.FindWithElasticsearchAsync(_elasticOptions.PublishedIndex, filter);
+        var result = await _contentService.FindWithElasticsearchAsync(useUnpublished ? _elasticOptions.UnpublishedIndex :_elasticOptions.PublishedIndex, filter);
```
Comment (jdtoombs) [inline api/net/Areas/Subscriber/Controllers/ContentController.cs:85]: @Fosol - just pinging to let you know this has been updated. Also wanted to see if this approach was ok?

**Review by Fosol — APPROVED** [review]

**Pull request merged**